Build failure: python312Packages.apprise

### Steps To Reproduce

Steps to reproduce the behavior:
I'm trying to use Healthchecks 3.6 (see PR #345283), but there seems to be a build failure (both [locally](https://github.com/diogotcorreia/dotfiles/actions/runs/11202665096/job/31139085431) and [in Hydra](https://hydra.nixos.org/build/274175356)).

### Build log

See hydra log: https://hydra.nixos.org/build/274162033

```
 error: builder for '/nix/store/4ia3amw0bpfm6klp201by36wdax2fb0i-python3.12-apprise-1.9.0.drv' failed with exit code 1;
       last 25 log lines:
       > test/test_persistent_store.py::test_persistent_storage_cache_object
       >   /build/apprise-1.9.0/test/conftest.py:83: ResourceWarning: unclosed file <_io.FileIO name='/build/pytest-of-nixbld/pytest-0/popen-gw2/test_persistent_custom_io0/default/tmp/tmp6wqi0a_m' mode='wb' closefd=True>
       >     gc.collect()
       >   Enable tracemalloc to get traceback where the object was allocated.
       >   See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
       >
       > test/test_persistent_store.py::test_persistent_storage_cache_object
       >   /build/apprise-1.9.0/test/conftest.py:83: ResourceWarning: unclosed file <_io.FileIO name='/build/pytest-of-nixbld/pytest-0/popen-gw2/test_persistent_custom_io0/default/tmp/tmpnc5w646l' mode='wb' closefd=True>
       >     gc.collect()
       >   Enable tracemalloc to get traceback where the object was allocated.
       >   See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > SKIPPED [1] test/test_plugin_dbus.py:48: Skipping dbus-python based tests
       > SKIPPED [1] test/test_plugin_syslog.py:42: Skipping syslog based tests
       > SKIPPED [1] test/test_apprise_translations.py:232: Unique Windows test cases
       > SKIPPED [1] test/test_asyncio.py:44: Requires Python 3.0 to 3.6
       > SKIPPED [1] test/test_plugin_fcm.py:905: Requires that cryptography NOT be installed
       > SKIPPED [1] test/test_plugin_growl.py:60: Requires that gntp NOT be installed
       > SKIPPED [1] test/test_plugin_mqtt.py:72: Requires that `paho-mqtt` is NOT installed
       > SKIPPED [1] test/test_plugin_windows.py:196: Requires win32api, win32con, and win32gui
       > SKIPPED [1] test/test_plugin_simplepush.py:120: Requires that cryptography NOT be installed
       > FAILED test/test_plugin_wxpusher.py::test_plugin_wxpusher_urls - AssertionError
       > ============ 1 failed, 505 passed, 9 skipped, 6 warnings in 34.41s =============
       For full logs, run 'nix log /nix/store/4ia3amw0bpfm6klp201by36wdax2fb0i-python3.12-apprise-1.9.0.drv'.
```

### Additional context

See these commits in my dotfiles: https://github.com/diogotcorreia/dotfiles/commit/00bcf3e6d061d06672426f8d86ddde93ddc4906e https://github.com/diogotcorreia/dotfiles/commit/2b1a7e4ac47c5f7870eb1ba38aca163ba665bcaf

`nixos-unstable` version: bc947f541ae55e999ffdb4013441347d83b00feb

### Notify maintainers

@getchoo since you are the package maintainer
@fabaff since you were the one to last update apprise in #345278

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.52, NixOS, 24.05 (Uakari), 24.05.20240925.759537f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.7`
 - nixpkgs: `/etc/channels/nixpkgs`
```

(^^ this is very useless because this is not the version that's failing)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Ugh, this just became a bit weirder. I was able to build this locally on my laptop without any errors :thinking: 

The derivation hash seems to be the same:

```
nix-repl> :b pkgs.unstable.python312Packages.apprise

This derivation produced the following outputs:
  dist -> /nix/store/qv9kfh9g9xs6c6ax642idy9wabcdm5in-python3.12-apprise-1.9.0-dist
  out -> /nix/store/pns9dzfv1ns6fl09741q5apfn9yr2gy2-python3.12-apprise-1.9.0

nix-repl> pkgs.unstable.python312Packages.apprise
«derivation /nix/store/4ia3amw0bpfm6klp201by36wdax2fb0i-python3.12-apprise-1.9.0.drv»
```

Logs (`nix log /nix/store/4ia3amw0bpfm6klp201by36wdax2fb0i-python3.12-apprise-1.9.0.drv`): https://bin.diogotc.com/qokukalufu

---

To confirm I'm not crazy, I deleted the store path and tried it again, and it seems that the test just isn't deterministic (this is the same repl session):

```
nix-repl> :b pkgs.unstable.python312Packages.apprise
error: builder for '/nix/store/4ia3amw0bpfm6klp201by36wdax2fb0i-python3.12-apprise-1.9.0.drv' failed with exit code 1;
       last 10 log lines:
       > SKIPPED [1] test/test_plugin_syslog.py:42: Skipping syslog based tests
       > SKIPPED [1] test/test_apprise_translations.py:232: Unique Windows test cases
       > SKIPPED [1] test/test_asyncio.py:44: Requires Python 3.0 to 3.6
       > SKIPPED [1] test/test_plugin_growl.py:60: Requires that gntp NOT be installed
       > SKIPPED [1] test/test_plugin_fcm.py:905: Requires that cryptography NOT be installed
       > SKIPPED [1] test/test_plugin_mqtt.py:72: Requires that `paho-mqtt` is NOT installed
       > SKIPPED [1] test/test_plugin_simplepush.py:120: Requires that cryptography NOT be installed
       > SKIPPED [1] test/test_plugin_windows.py:196: Requires win32api, win32con, and win32gui
       > FAILED test/test_plugin_wxpusher.py::test_plugin_wxpusher_result_set - IndexError: list index out of range
       > ============ 1 failed, 505 passed, 9 skipped, 6 warnings in 40.00s =============
       For full logs, run 'nix log /nix/store/4ia3amw0bpfm6klp201by36wdax2fb0i-python3.12-apprise-1.9.0.drv'.

nix-repl> :b pkgs.unstable.python312Packages.apprise

This derivation produced the following outputs:
  dist -> /nix/store/qv9kfh9g9xs6c6ax642idy9wabcdm5in-python3.12-apprise-1.9.0-dist
  out -> /nix/store/pns9dzfv1ns6fl09741q5apfn9yr2gy2-python3.12-apprise-1.9.0
```

--- a/pkgs/development/python-modules/apprise/default.nix
+++ b/pkgs/development/python-modules/apprise/default.nix
@@ -64,6 +64,9 @@ buildPythonPackage rec {
     "test_plugin_mqtt_general"
     # Nondeterministic. Fails with `assert 3 == 2`
     "test_plugin_matrix_transaction_ids_api_v3"
+    # Nondeterministic. Fails with `IndexError`
+    # https://github.com/NixOS/nixpkgs/issues/346894
+    "test_plugin_wxpusher_result_set"
     # Nondeterministic. Fails with `AssertionError`
     "test_plugin_xbmc_kodi_urls"
     # Nondeterministic. Fails with `AssertionError`

